### PR TITLE
Saner defaults for peak_local_max, untested.

### PR DESCRIPTION
**Pull request by anntzer**
First attempt at fixing #1246.

### CODE REVIEW

File: skimage/feature/peak.py [hunk 1]
```diff
@@ -138,12 +139,16 @@ def peak_local_max(image, min_distance=10, threshold_abs=0, threshold_rel=0.1,
...
+                      else 2 * min_distance)
+            image[:remove // 2] = 0
+            image[-remove // 2:] = 0
             image = image.swapaxes(0, i)
 
     # find top peak candidates above a threshold
-    peak_threshold = max(np.max(image.ravel()) * threshold_rel, threshold_abs)
+    peak_threshold = threshold_abs
+    if threshold_rel is not None:
+        peak_threshold = max(peak_threshold, image.max())
```
Comment (tonysyu) [inline skimage/feature/peak.py]: You dropped `threshold_rel` from the relative-max calculation (i.e. `image.max() * threshold_rel`).

File: skimage/feature/peak.py [hunk 2]
```diff
@@ -138,12 +139,16 @@ def peak_local_max(image, min_distance=10, threshold_abs=0, threshold_rel=0.1,
...
+            remove = (footprint.shape[i] if footprint is not None
+                      else 2 * min_distance)
+            image[:remove // 2] = 0
+            image[-remove // 2:] = 0
             image = image.swapaxes(0, i)
 
     # find top peak candidates above a threshold
-    peak_threshold = max(np.max(image.ravel()) * threshold_rel, threshold_abs)
+    peak_threshold = threshold_abs
+    if threshold_rel is not None:
```
Comment (tonysyu) [inline skimage/feature/peak.py]: Instead of defaulting to `None`, you could, instead, set `threshold_rel=1` by default, and get rid of this conditional.

**Comment by anntzer** [comment]
The latest commit is basically where I was when I mentioned that adding `threshold_rel=0` everywhere to the tests is not really a solution.  It does also fix the handling of `threshold_rel`, just differently from how @tonysyu suggested.

**Comment by tonysyu** [comment]
@stefanv: Oh, I guess you were pinging me to respond to @emmanuelle's question, not to review the PR. I don't think I wrote that test, but I could be wrong. Either way, I agree it doesn't really make much sense. (Unless we were taking the centroid of peaks within the min-distance/footprint, but I don't think we are.)

**Comment by cdeil** [comment]
There's a [find_peaks](http://photutils.readthedocs.org/en/latest/_modules/photutils/detection/core.html#find_peaks) function in `photutils` ... it was a `peak_local_max` wrapper at some point but now just uses `scipy.ndimage`. I'm just mentioning it because it could be a useful example for API / implementation.

**Comment by ahojnnes** [comment]
What's the status here?

**Comment by ahojnnes** [comment]
@stefanv @cdeil @emmanuelle @JDWarner @jni Do all agree that the new defaults are better and/or good enough to be merged? If so, I'll just try to take the current status and open a new mergable PR.

**Comment by JDWarner** [comment]
They're better, and we need to improve this. I'm less enthusiastic about the changes needed to all of the tests to make this possible, though.

**Comment by stefanv** [comment]
Thanks, @ahojnnes, it would be great if you could get the tests passing with the new defaults.

**Comment by ahojnnes** [comment]
Closed in favor of https://github.com/scikit-image/scikit-image/pull/1898. Thanks @anntzer for your contribution and sorry for the delay. I'll take it from here to get it finally merged!

**Comment by emmanuelle** [comment]
So, this PR should be closed, right @ahojnnes ?

**Comment by JDWarner** [comment]
Yes, @emmanuelle - I missed this one when merging #1898.

**Pull request closed without merging**